refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java

```diff
@@ -246,27 +223,20 @@ private Problem reallyUndoRefactoring(boolean saveAfterDone) {
                     f.undoChange();
                 }
             }
-            for (Transaction commit : SPIAccessor.DEFAULT.getCommits(bag)) {
-                SPIAccessor.DEFAULT.check(commit, true);
-            }
-            UndoableWrapper wrapper = MimeLookup.getLookup("").lookup(UndoableWrapper.class);
+            SPIAccessor.DEFAULT.getCommits(bag).forEach(commit -> SPIAccessor.DEFAULT.check(commit, true));
+            UndoableWrapper undoableWrapper = MimeLookup.getLookup("").lookup(UndoableWrapper.class);
             for (ListIterator<Transaction> commitIterator = commits.listIterator(commits.size()); commitIterator.hasPrevious();) {
                 final Transaction commit = commitIterator.previous();
-                setWrappers(commit, wrapper);
+                undoableWrapper.setActive(true, this);
                 commit.rollback();
-                unsetWrappers(commit, wrapper);
-            }
-            wrapper.close();
-            for (Transaction commit : SPIAccessor.DEFAULT.getCommits(bag)) {
-                SPIAccessor.DEFAULT.sum(commit);
+                undoableWrapper.setActive(false, null);
```

Comment:
Similar to before, should `undoableWrapper.setActive(false, null);` be within a finally block?

Reply:
It was not before. Probably it will be of use, if you explain it to me.

Reply:
Sorry, I was unclear. By "similar to before", I meant "similar to [my previous comment](https://github.com/apache/incubator-netbeans/pull/361#discussion_r161065679) up above".

Reply:
My apologies. I'm not able to answer that question. I would add things without concrete knowledge about what I'm actually doing.

Reply:
Which is very true.